### ISSUE TITLE
RegexNotFoundError: Unable to extract uploader id

### ISSUE THREAD

**Issue by starling80** [ytdl-org/youtube-dl#31742]
- [x ] I'm reporting a broken site support issue
- [x ] I've verified that I'm running youtube-dl version **2021.12.17**
- [x ] I've checked that all provided URLs are alive and playable in a browser
- [x ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x ] I've searched the bugtracker for similar bug reports including closed ones
- [x ] I've read bugs section in FAQ

## Verbose log
```
/usr/local/bin/youtube-dl --verbose https://www.youtube.com/watch?v=_VONMkKkdf4
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--verbose', u'https://www.youtube.com/watch?v=_VONMkKkdf4']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 2.7.18 (CPython) - Linux-5.4.0-139-generic-x86_64-with-LinuxMint-20.2-uma
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] _VONMkKkdf4: Downloading webpage
ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1794, in _real_extract
    'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id') if owner_profile_url else None,
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
RegexNotFoundError: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.



```

## Description

I cannot download the two following links:
https://www.youtube.com/watch?v=_VONMkKkdf4
https://www.youtube.com/watch?v=BasMCUEaawg
The problem started at date: 2023-03-01. I think it could be an issue of protection, since I still can download old links, which worked several days before. Could it be, that the youtube's protection mechanisms changed around that date?

yes I checked several similar bug posts but I am not experienced enough to find a suitable solution by myself. SRY!
It seems like "RegexNotFoundError: Unable to extract uploader id" is the root of the problem, but I am nowhere near to be qualified enough to analyze the problem...

Any HELP, IDEAS, THOUGHTS & WORKAROUNDS are appreciated! Many THANKS!

I got my latest youtube-dl version via wget, like follows:
sudo wget https://yt-dl.org/downloads/latest/youtube-dl -O /usr/local/bin/youtube-dl

**Comment by estevan3** [comment]
I am having exactly the same problem!

**Comment by Belatar** [comment]
+1

> [debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['https://www.youtube.com/watch?v=7HKiAJMx084', '-F', '--verbose']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.22621
[debug] exe versions: none
[debug] Proxy map: {}
[youtube] 7HKiAJMx084: Downloading webpage
ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\YoutubeDL.py", line 815, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\YoutubeDL.py", line 836, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 534, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\youtube.py", line 1794, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 1012, in _search_regex
youtube_dl.utils.RegexNotFoundError: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

**Comment by Guillevadela** [comment]
I have the same problem with all the URL's 

here it is the error on my terminal:
youtube-dl -f bestvideo+bestaudio https://www.youtube.com/watch\?v\=nQobb4pcU78\&t\=4670shttps://www.youtube.com/watch\?v\=Ym2pxzWpTNw\&t\=75s
[youtube] nQobb4pcU78: Downloading webpage
ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

**Comment by WafflesMcDuff** [comment]
This is a duplicate of #31530

**Comment by estevan3** [comment]
This has already been solved! Please read the top of [this thread](https://github.com/ytdl-org/youtube-dl/issues/31530), the change you need to make is [here](https://github.com/yt-dlp/yt-dlp/commit/149eb0bbf34fa8fdf8d1e2aa28e17479d099e26b):

> In any case the extraction of an optional field shouldn't crash the extraction:
> 
> ```diff
> -            'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id') if owner_profile_url else None,
> +            'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id', fatal=False) if owner_profile_url else None,
> ```

**Comment by WafflesMcDuff** [comment]
The other option is to just download the latest nightly build from https://github.com/ytdl-patched/youtube-dl/releases

This version of ytdl hasn't been maintained since 2021

**Comment by starling80** [comment]
thank you very much!
so, "wget - latest" does not get me the latest version???

**Comment by starling80** [comment]
OK it works now!
I am using now this source:
sudo wget https://github.com/ytdl-patched/youtube-dl/releases/download/2023.03.02.334/youtube-dl -O /usr/local/bin/youtube-dl

thank you very much!

**Comment by dirkf** [comment]
Read #30839!
